### PR TITLE
feat: add link card component to home page

### PR DESCRIPTION
**Pull request by sowmyadev-kota**
#### Description of changes:

This feature will add link-card-component to the home page

Staging site: https://link-card-component-home.d21cx590oj6ufl.amplifyapp.com/
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/pages/index.tsx [hunk 1]
```diff
@@ -6,6 +6,12 @@ import { GetStartedPopover } from '@/components/GetStartedPopover';
 import { IconChevron } from '@/components/Icons';
 
 import { trackPageVisit } from '@/utils/track';
+import LinkCard from '@/components/LinkCard';
```
Comment (reesscot) [inline src/pages/index.tsx:9]: @hbuchel Should this be in that file that allows components to be used anywhere?

File: src/pages/index.tsx [hunk 2]
```diff
@@ -69,7 +75,36 @@ export default function Page() {
...
+          icon={() => <IconDiscord fontSize="2rem" />}
+        >
+          {'Amplify Discord'}
+        </LinkCard>
+        <LinkCard
+          isExternal={true}
+          href={''}
+          icon={() => <IconAmplify fontSize="2rem" />}
+        >
+          {"What's next for Amplify"}
```
Comment (reesscot) [inline src/pages/index.tsx:98]: brackets and quotes are unnecessary
```suggestion
What's next for Amplify
```

File: src/pages/index.tsx [hunk 3]
```diff
@@ -69,7 +75,36 @@ export default function Page() {
...
         </Text>
       </Flex>
-
+      <LinkCardCollection>
+        <LinkCard
+          isExternal={true}
+          href={''}
+          icon={() => <IconGithub fontSize="2rem" />}
+        >
+          {'JavaScript Libraries on Github'}
```
Comment (reesscot) [inline src/pages/index.tsx:84]: Don't these need to be variable based on the platform selected?

Comment (sowmyadev-kota) [reply src/pages/index.tsx:84]: @hbuchel   , please let me know!

File: src/pages/index.tsx (hunk 1 above)
Comment (hbuchel) [reply src/pages/index.tsx:9]: I don't see this on the designs for "inner" pages so I think it's ok to just leave out from that for now.

File: src/styles/link-card-styles.scss [hunk 4]
```diff
@@ -0,0 +1,26 @@
+.link-card {
```
Comment (reesscot) [inline src/styles/link-card-styles.scss:1]: nit: this file can just be named link-card, consistent with the other scss files in the folder

File: src/pages/index.tsx (hunk 2 above)
Comment (reesscot) [reply src/pages/index.tsx:98]: Same feedback as above

File: src/pages/[platform]/index.tsx [hunk 5]
```diff
@@ -38,6 +49,21 @@ export function getStaticProps(context) {
 const PlatformOverview = ({ platform }) => {
   return (
     <>
+      <h1>// Amplify docs for {PLATFORM_DISPLAY_NAMES[platform]}</h1>
+      <LinkCardCollection>
+        <LinkCard isExternal={true} href="" icon={<IconGithub />}>
+          {PLATFORM_DISPLAY_NAMES[platform]} Libraries on Github
+        </LinkCard>
+        <LinkCard isExternal={true} href="" icon={<IconDiscord />}>
+          {'Amplify Discord'}
```
Comment (reesscot) [inline src/pages/[platform]/index.tsx:58]: Same comment about just using a string here

File: src/pages/index.tsx [hunk 6]
```diff
@@ -73,7 +79,36 @@ export default function Page() {
         </Text>
         <FrameworkGrid currentKey="javascript" />
       </Flex>
-
+      <LinkCardCollection>
+        <LinkCard
+          isExternal={true}
+          href={''}
```
Comment (reesscot) [inline src/pages/index.tsx:85]: Can we please add the actual URLS here

File: src/styles/styles.scss [hunk 7]
```diff
@@ -24,3 +24,4 @@
 [id*='lastupdated-'] {
   display: none;
 }
+@import './link-card.css';
```
Comment (reesscot) [inline src/styles/styles.scss:27]: Please place this above organized with the other primitives in alphabetical order

**Review by hbuchel — APPROVED** [review]

File: src/components/FeatureLists/FeatureLists.tsx [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import { Grid } from '@aws-amplify/ui-react';
+const FeatureLists = ({ children, heading, level }) => {
```
Comment (reesscot) [inline src/components/FeatureLists/FeatureLists.tsx:2]: Not all these props are being used

**Review by timngyn — APPROVED** [review]

**Review by jacoblogan — DISMISSED** [review]

**Review by reesscot — COMMENTED** [review]
Left some more thoughts

File: src/pages/index.tsx [hunk 9]
```diff
@@ -73,7 +81,36 @@ export default function Page() {
...
+          icon={() => <IconAmplify fontSize="2rem" />}
+        >
+          What's next for Amplify
+        </LinkCard>
+        <LinkCard
+          isExternal={true}
+          href={''}
```
Comment (timngyn) [inline src/pages/index.tsx:108]: For Learn, the URL would be `https://amplify.aws/learn/`

File: src/pages/index.tsx [hunk 10]
```diff
@@ -73,7 +81,36 @@ export default function Page() {
...
+          icon={() => <IconGithub fontSize="2rem" />}
+        >
+          JavaScript Libraries on Github
+        </LinkCard>
+        <LinkCard
+          isExternal={true}
+          href={''}
```
Comment (timngyn) [inline src/pages/index.tsx:94]: For Discord, I think this would be the URL `https://discord.com/invite/amplify`

**Review by hbuchel — APPROVED** [review]

**Review by timngyn — APPROVED** [review]

**Pull request merged**